doc/api/cluster.markdown

```diff
@@ -209,40 +551,42 @@ in the master process using the message system:
 
 ### Event: 'online'
 
-Similar to the `cluster.on('online')` event, but specific to this worker.
+Triggered only within the master process when a specific worker is online
+and ready to receive messages.
 
     cluster.fork().on('online', () => {
       // Worker is online
     });
 
-It is not emitted in the worker.
-
 ### worker.disconnect()
 
-In a worker, this function will close all servers, wait for the `'close'` event on
-those servers, and then disconnect the IPC channel.
+When called from within a worker process, the `worker.disconnect()` method
+will close all listening servers, wait for the `'close'` event on those servers
+to be triggered, and will then disconnect the IPC channel with the master
+process.
 
-In the master, an internal message is sent to the worker causing it to call
-`.disconnect()` on itself.
+When called on the master process, an internal control message is sent to
+the worker instructing it to call the `worker.disconnect()` message itself.
 
-Causes `.suicide` to be set.
+When called, the `worker.suicide` flag will be set to `true`.
 
-Note that after a server is closed, it will no longer accept new connections,
-but connections may be accepted by any other listening worker. Existing
-connections will be allowed to close as usual. When no more connections exist,
-see [server.close()][], the IPC channel to the worker will close allowing it to
-die gracefully.
+Note that after a server is closed, new connections will no longer be accepted
+by that worker, but connections may be accepted by any other listening worker.
+Existing connections will be allowed to close as usual. When no more
+connections exist (see [server.close()][]) the IPC channel to the worker will
+close allowing it to exit gracefully.
 
-The above applies *only* to server connections, client connections are not
-automatically closed by workers, and disconnect does not wait for them to close
-before exiting.
+Client connections, on the other hand, are not automatically closed by workers,
+and disconnect does not wait for client connections to close before exiting.
 
-Note that in a worker, `process.disconnect` exists, but it is not this function,
-it is [`disconnect`][].
+Note that in a worker process, the `worker.disconnect()` and
+`process.disconnect()` methods are *not* equivalent to one another. Worker
```

Comment:
This surprises me, cluster still does. I don't have time to read the source, but I would verify this is true.